packages/components/src/color-palette/index.tsx

```diff
@@ -257,43 +252,50 @@ function UnforwardedColorPalette(
 					isRenderedInSidebar={ __experimentalIsRenderedInSidebar }
 					renderContent={ renderCustomColorPicker }
 					renderToggle={ ( { isOpen, onToggle } ) => (
-						<Flex
-							as={ 'button' }
-							ref={ customColorPaletteCallbackRef }
-							justify="space-between"
-							align="flex-start"
-							className="components-color-palette__custom-color"
-							aria-expanded={ isOpen }
-							aria-haspopup="true"
-							onClick={ onToggle }
-							aria-label={ customColorAccessibleLabel }
-							style={
-								showTransparentBackground( value )
-									? { color: '#000' }
-									: {
-											background: value,
-											color:
-												colordColor.contrast() >
-												colordColor.contrast( '#000' )
-													? '#fff'
-													: '#000',
-									  }
-							}
-						>
-							<FlexItem
-								isBlock
-								as={ Truncate }
-								className="components-color-palette__custom-color-name"
-							>
-								{ buttonLabelName }
-							</FlexItem>
-							<FlexItem
-								as="span"
-								className="components-color-palette__custom-color-value"
+						<VStack spacing={ 0 }>
+							<button
+								ref={ customColorPaletteCallbackRef }
+								className="components-color-palette__custom-color-button"
+								aria-expanded={ isOpen }
+								aria-haspopup="true"
+								onClick={ onToggle }
+								aria-label={ customColorAccessibleLabel }
+								style={
+									showTransparentBackground( value )
+										? {}
+										: {
+												background: value,
+										  }
+								}
+							/>
+							<VStack
+								className="components-color-palette__custom-color-text-wrapper"
+								spacing={ 0.5 }
 							>
-								{ valueWithoutLeadingHash }
-							</FlexItem>
-						</Flex>
+								<Truncate className="components-color-palette__custom-color-name">
+									{ value
+										? buttonLabelName
+										: 'No color selected' }
+								</Truncate>
+								{ /*
+								This `Truncate` is always rendered, even if
+								there is no `displayValue`, to ensure the layout
+								does not shift
+								*/ }
+								<Truncate
+									className={ classnames(
+										'components-color-palette__custom-color-value',
+										{
+											'components-color-palette__custom-color-value--is-hex':
+												displayValue?.startsWith( '#' ),
+										}
+									) }
+									style={ { minHeight: '18px' } }
```

Comment:
I wound up placing this `min-height` inline because in the editor it was getting zeroed out when it was included in the component's styles. This wasn't an issue in Storybook, I think the editor is loading things in a different order. Happy to change this is there's a better approach though.

Reply:
Marco's pseudo-element approach sounds best to me as well. (For completeness, another possible single-property workaround is `flex-basis: 18px`.)

## `min-height` issue

The [conflicting style](https://github.com/WordPress/gutenberg/blob/2944062507bbdf6a05e8d56f9f18aa24129c955f/packages/components/src/flex/styles.ts#L36) is coming from the parent `VStack`, which was [added by Q](https://github.com/ItsJonQ/g2/pull/149) as workaround for a [flexbox quirk](https://stackoverflow.com/a/36247448). The statement in the PR description "This shouldn't affect potential child styles as the universal `*` selector has a specificity score of `0`" is unfortunately not true in practice, because Emotion will output this as `.some-selector > *`, which has a specificity score of 1.

## Load order issue

Ultimately, I'm afraid we don't have much control over the load order of Emotion/stylesheet styles in consumer environments, since the consumer is responsible for loading the wp-components stylesheet. We could possibly match the order of the Storybook and Gutenberg, but that may lead to a false sense of security. I guess the best way is to stick with a single CSS system 😅

Reply:
Thank you for the sleuthing, @mirka ! Let's go for the pseudo-element solution.

Hopefully the style conflict will be implicitly solved if/when we migrate to a different styling toolchain — until then, we have to be aware of the quirk